utils.go

```diff
@@ -57,29 +73,38 @@ func bigIntArrayToStringArray(array []*big.Int) []string {
 	return res
 }
 
-func getSlots(claim *core.Claim) []*big.Int {
-	inputs := make([]*big.Int, 0)
+type nodeAuxValue struct {
+	key   *merkletree.Hash
+	value *merkletree.Hash
+	noAux string
+}
 
-	index, value := claim.RawSlots()
+func getNodeAuxValue(a *merkletree.NodeAux) nodeAuxValue {
 
-	for i := range index {
-		inputs = append(inputs, index[i].ToInt())
+	aux := nodeAuxValue{
+		key:   &merkletree.HashZero,
+		value: &merkletree.HashZero,
+		noAux: "1",
 	}
-	for i := range value {
-		inputs = append(inputs, value[i].ToInt())
+
+	if a != nil && a.Value != nil && a.Key != nil {
+		aux.key = a.Key
+		aux.value = a.Value
+		aux.noAux = "0"
 	}
-	return inputs
-}
 
-func convertMTPtoProof(claimEntryMTP *merkletree.Proof) Proof {
-	var claimEntryProof Proof
-	claimEntryProof.Siblings = claimEntryMTP.AllSiblings()
+	return aux
+}
 
-	if claimEntryMTP.NodeAux != nil {
-		claimEntryProof.NodeAux = &NodeAux{
-			HIndex: claimEntryMTP.NodeAux.Key,
-			HValue: claimEntryMTP.NodeAux.Key,
-		}
+func IDFromStr(s string) (*core.ID, error) {
```

Comment:
Name of this function is ambiguous, I thought it converts base58 encoded ID, not base 10. Because there's `core.ID.String()` & `core.IDFromString()` which are doing base58 encoding and decoding respectively. `IDFromIntStr` or something similar would be better.

Reply:
+1